core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2420,6 +2420,44 @@ private[spark] object Utils extends Logging {
   }
 }
 
+private[spark] class CallerContext(
+   appName: Option[String] = None,
+   appID: Option[String] = None,
+   appAttemptID: Option[String] = None,
+   jobID: Option[Int] = None,
+   stageID: Option[Int] = None,
+   stageAttemptId: Option[Int] = None,
+   taskId: Option[Long] = None,
+   taskAttemptNumber: Option[Int] = None) extends Logging {
+
+   val AppName = if (appName.isDefined) s"_AppName_${appName.get}" else ""
```

Comment:
I have updated the PR to remove appName, and replace appName with something to differentiate the context from ApplicationMaster vs Yarn Client vs Task. But for AppID, I think it is better to keep it since in hdfs-audit.log, there is no info about application. For example, the record below was produced when Task did a read/write operation to HDFS, except `callerContext`, there is no other info about application:
```
2016-09-14 22:29:06,526 INFO FSNamesystem.audit: allowed=true   ugi=wyang (auth:SIMPLE) ip=/127.0.0.1   cmd=open    src=/lr_big.txt dst=null    perm=null   proto=rpc   callerContext=SPARK_AppID_application_1473908768790_0007_JobID_0_StageID_0_0_TaskId_2_0
```